src/rules/prefer-native-locators.ts

```diff
@@ -0,0 +1,139 @@
+import { getStringValue, isPageMethod } from '../utils/ast'
+import { createRule } from '../utils/createRule'
+
+export default createRule({
+  create(context) {
+    return {
+      CallExpression(node) {
+        const { testIdAttribute } = {
+          testIdAttribute: 'data-testid',
+          ...((context.options?.[0] as Record<string, unknown>) ?? {}),
+        }
+
+        if (node.callee.type !== 'MemberExpression') return
+        const method = getStringValue(node.callee.property)
+        const query = getStringValue(node.arguments[0])
+        const isLocator = isPageMethod(node, 'locator') || method === 'locator'
```

Comment:
Why the OR statement here?

Reply:
I copied this from the `no-raw-locators` rule. Probably not super important to account for calls like `this.locator()` right now. I went ahead and removed this in https://github.com/playwright-community/eslint-plugin-playwright/pull/308/commits/a93d0ebde3b65056ff4da0eb5aa85c7336707d84